File: TenMinsJava/day83.java
// deleteCSVRow(rowId) deletes a row from csv specified by the rowId
package TenMinsJava;

import java.io.File;
import java.io.FileWriter;
import java.util.ArrayList;
import java.util.Scanner;

public class day83 {
    public static void deleteCSVRow(int rowId) {
        try {
            File file = new File("D:/coding30minsAday/textFiles/JTrialCsv.csv");
            Scanner fScan = new Scanner(file);
            ArrayList<String> lines = new ArrayList<>();

            // Step 1: Read all lines
            while (fScan.hasNextLine()) {
                lines.add(fScan.nextLine());
            }
            fScan.close();

            // modifies the value
            lines.remove(rowId - 1);

            // rewrite the csv
            FileWriter w = new FileWriter(file);
            for (String string : lines) {
                w.write(string + "\n");
            }
            w.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: TenMinsPy/day83.py
# deleteCSVRow(rowId) deletes a row from csv specified by the rowId

import csv

def deleteCSVRow(row_id):
    try:
        # Step 1: Read all lines
        file_path="D:/coding30minsAday/textFiles/JTrialCsv.csv"
        with open(file_path, "r") as f:
            lines = f.readlines()

        # Step 2: Modify the value
        if len(lines) >= row_id:
            lines.pop(row_id - 1)

        # Step 3: Rewrite the CSV
        with open(file_path, "w") as f:
            f.writelines(lines)

    except Exception as e:
        print("Error:", e)